scripts/setup

```diff
@@ -5,16 +5,23 @@ for tool in "${@}"; do
   echo "Installing '$tool'..."
   export tool
   install_method="$(jq -r '.[env.tool].method' dev-tools.json)"
+  target="$HOME/.local/bin/$tool"
+  export target
   echo "  install_method: $install_method"
   case "$install_method" in
+  "apt")
+    sudo apt-get install -yy "${tool}"
+    ;;
   "curl")
-    url="$(jq -r '.[env.tool].url' dev-tools.json)"
-    target="$HOME/.local/bin/$tool"
+    url="$(jq -r '.[env.tool].version as $version | .[env.tool].url | gsub("[$]{VERSION}"; $version)' dev-tools.json)"
+    pipe=($(jq -r '.[env.tool].pipe // "cat"' dev-tools.json))
     mkdir -p "$(dirname "$target")"
-    curl -Lf "$url" | install -m 755 /dev/stdin "$target"
+    curl -Lf "$url" | "${pipe[@]}" | install -m 755 /dev/stdin "$target"
     ;;
   "sh")
-    version="$(jq -r '.[env.tool].version' dev-tools.json)" "$0-$tool"
+    (
+      version="$(jq -r '.[env.tool].version' dev-tools.json)" "$0-$tool"
+    )
```

Comment:
The parenthentheses create a subshell that protects the setup command from install scripts that change the environment, e.g. by changing directory.  Any changes are contained within the subshell.